src/Bicep.Core/Analyzers/Linter/LinterAnalyzer.cs

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Bicep.Core.Analyzers.Interfaces;
+using Bicep.Core.Semantics;
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Globalization;
+using System.Linq;
+using System.Reflection;
+
+namespace Bicep.Core.Analyzers.Linter
+{
+    internal class LinterAnalyzer : IBicepAnalyzer
+    {
+        public static string AnalyzerName => "Bicep Internal Linter";
+        public static readonly bool IsCliInvoked;
+
+        private static readonly IEnumerable<Type> RuleTypes;
+        private static readonly ImmutableArray<IBicepAnalyzerRule> RuleSet;
+
+        static LinterAnalyzer()
+        {
+            var entryAssembly = Assembly.GetEntryAssembly();
+            IsCliInvoked = entryAssembly.Location.EndsWith("bicep.dll", true, CultureInfo.CurrentCulture);
+            
+            RuleTypes = Assembly.GetExecutingAssembly()
+                .GetTypes()
+                .Where(t => t.IsSubclassOf(typeof(LinterRule)));
+
+            RuleSet = CreateLinterRules().ToImmutableArray();
+        }
+
+        internal static  IEnumerable<IBicepAnalyzerRule> CreateLinterRules()
+        {
+            foreach(var ruleType in RuleTypes)
+            {
+                if(typeof(IBicepAnalyzerRule).IsAssignableFrom(ruleType))
+                {
+                    yield return (IBicepAnalyzerRule) Activator.CreateInstance(ruleType);
+                }
+            }
+        }
+
+        public IEnumerable<IBicepAnalyzerRule> GetRuleSet() => RuleSet;
+
+        public IEnumerable<IBicepAnalyzerDiagnostic> Analyze(SemanticModel semanticModel)
```

Comment:
>Analyze [](http://example.com/codeflow?start=53&length=7)

Btw, what's the plan for hardening against badly written analyzers? Right now any exception will take down the language server. (VS code will restart it a few times but it will pass the same document every time, so it will keep crashing assuming the defect is deterministic.)

Realistically, I don't think we need to do more than catch exceptions and validate all data that comes out of an analyzer, but it can probably wait until we have linter extensibility.

Reply:
I feel like it might be worth adding this now, and possibly also hooking it up to the telemetry pipeline (or to the 'report an issue' button) so that we can get some data if analyzers aren't working as expected. It's going to be quite difficult to exhaustively test all the code paths in analyzers.

Reply:
I think this is desired for MVP but can wait for a separate PR.

Reply:
https://github.com/jfleisher/bicep/pull/14